near-sdk/src/store/lookup_set/mod.rs

```diff
@@ -118,7 +210,24 @@ where
     /// [`Drop`]ed. This will write all modified values to storage but keep all cached values
     /// in memory.
     pub fn flush(&mut self) {
-        self.map.flush()
+        let mut buf = Vec::new();
+        for (k, v) in self.cache.borrow_mut().iter_mut() {
+            match v.get() {
+                EntryState::Inserted => {
+                    buf.clear();
+                    let lookup_key = Self::lookup_key(&self.prefix, k, &mut buf);
+                    env::storage_write(&lookup_key, &[]);
```

Comment:
So I'm pretty sure `&[]` serializes the length (0) so should use `()` in place of these

Reply:
I think you are confusing de-serialization with plain `storage_write` usage here. `env::storage_write` accepts `&[u8]` as the second parameter, so you can't pass `()` there anyway.